my_enumerable_module.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+# Extending Enumerable module with replicated methods for the following:
+# each, each_with_index, select, all?, none?, any?, count, map and inject
+#
+module Enumerable
+  def my_each
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i])
+    end
+  end
+
+  def my_each_with_index
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i], i)
+    end
+  end
+
+  def my_select
+    arr = self # this is to prevent a redundant-use-of-self error
+    result = []
+    0.upto(arr.length - 1) do |i|
+      result << arr[i] if yield(arr[i])
+    end
+    result
+  end
+
+  def my_all?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = true
+    0.upto(arr.length - 1) do |i|
+      response = false unless yield(arr[i])
+    end
+    response
+  end
+
+  def my_any?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = false
+    0.upto(arr.length - 1) do |i|
+      response = true if yield(arr[i])
+    end
+    response
+  end
+
+  def my_none?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = true
+    0.upto(arr.length - 1) do |i|
+      response = false if yield(arr[i])
+    end
+    response
+  end
+
+  def my_count(single_value = nil)
+    arr = self
+    counter = 0
+    if single_value
+      arr.my_each { |val| counter += 1 if val == single_value }
+    else
+      arr.my_each { |val| counter += 1 if yield(val) }
+    end
+    counter
+  end
+
+  def my_map(a_proc = nil)
+    arr = self # this is to prevent a redundant-use-of-self error
+    if block_given? && a_proc.nil?
+      arr.my_each_with_index { |v, i| arr[i] = yield(v) }
+    else
+      arr.my_each_with_index { |v, i| arr[i] = a_proc.call(v) }
+    end
+    arr
+  end
+
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def my_inject(init = 0, oper = :+)
+    obj = self
+    if init.is_a? Symbol
+      oper = init
+      init = 0
+    end
+    accum = obj[init]
+    if block_given?
+      (init + 1).upto(obj.length - 1) { |i| accum = yield(accum, obj[i]) }
+    else
+      (init + 1).upto(obj.length - 1) { |i| accum = accum.send(oper, obj[i]) }
+    end
+    accum
+  end
+  # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

Reply:
Corrected